oak/proto/introspection_events.proto

```diff
@@ -0,0 +1,105 @@
+//
+// Copyright 2019 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+syntax = "proto3";
+
+import "google/protobuf/timestamp.proto";
+
+package oak.event;
+
+// This messages defines an event sent by the Oak Runtime to allow for dynamic
+// introspection in non-production enviroments. It provides a "schema" to
+// describe the creation/destruction of nodes, channels, handles, as well as the
+// queuing of messages.
+message Event {
+  google.protobuf.Timestamp timestamp = 1;
```

Comment:
See https://github.com/protocolbuffers/protobuf/blob/master/src/google/protobuf/timestamp.proto on details for the type. If preferred we could also use a simple `int64` and insert a UNIX timestamp ourselves.

Timestamps would be useful, since they allow for "replaying" of events. I know precise timestamps can pose security risks, but these events are only sent in non-prod enviroments.